src/e2e/pages/dashBoardPage.ts

```diff
@@ -190,4 +225,39 @@ export class DashboardPage {
     await this.faqsPageLink.click();
     await this.page.waitForURL("**/firefox-monitor-faq");
   }
+
+  async verifyPremiumUpsellModalOptions() {
+    // verify subscription dialog elements
+    await expect(this.subscribeDialogCloseButton).toBeVisible();
+    await expect(this.yearlyMonthlyTablist).toBeVisible();
+
+    // verify user purchase choices
+    await this.yearlyTab.click();
+    await expect(this.subscribeDialogSelectYearlyPlanLink).toBeVisible();
+
+    await this.monthlyTab.click();
+    await expect(this.subscribeDialogSelectMonthlyPlanLink).toBeVisible();
+
+    // Check monthly redirection
+    await this.subscribeDialogSelectMonthlyPlanLink.click();
+    const purchasePage = new PurchasePage(this.page);
+    await purchasePage.subscriptionHeader.waitFor();
+    await expect(purchasePage.planDetails).toContainText("monthly");
+    await expect(purchasePage.subscriptionHeader).toBeVisible();
+
+    // Check yearly redirection
+    await this.page.goto(`${process.env.E2E_TEST_BASE_URL}`);
+    await this.subscribeButton.waitFor();
+    await this.subscribeButton.click();
+    await this.subscribeDialogSelectYearlyPlanLink.click();
+    await purchasePage.subscriptionHeader.waitFor();
+    // Text contains invisible characters which need to be hardcoded in an assertion
```

Comment:
Q: What does this comment mean? I see we have the `removeUnicodeChars(text: string): string {...}` below, but we aren't using it here to remove any invisible chars.

Reply:
Using the function now and removed the comment, https://github.com/mozilla/blurts-server/pull/4543/commits/890aec5177df521b278422223a7db4aced5ae7ab#diff-75ef62711cbcf8a457647f69a04069d3de68ce6e40887e0ac2373ee64ce2651bL253-L257